src/highdicom/map/sop.py

```diff
@@ -0,0 +1,789 @@
+from collections import defaultdict
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+from enum import Enum
+
+import numpy as np
+from pydicom.encaps import decode_data_sequence, encapsulate
+from pydicom.pixel_data_handlers.util import get_expected_length
+from highdicom.base import SOPClass
+from highdicom.content import (
+    PixelMeasuresSequence,
+    PlaneOrientationSequence,
+    PlanePositionSequence,
+)
+from highdicom.enum import CoordinateSystemNames
+from highdicom.frame import encode_frame
+from highdicom.map.content import RealWorldValueMapping
+from highdicom.seg.content import DimensionIndexSequence
+from highdicom.valuerep import check_person_name
+from pydicom import Dataset
+from pydicom.uid import (
+    UID,
+    ExplicitVRLittleEndian,
+    ImplicitVRLittleEndian,
+    JPEG2000Lossless,
+    RLELossless,
+)
+
+
+class _PixelDataType(Enum):
+    """Helper enum for tracking the type of the pixel data"""
+
+    SHORT = 1
+    USHORT = 2
+    SINGLE = 3
+    DOUBLE = 4
+
+
+class ParametricMap(SOPClass):
+
+    """SOP class for a Parametric Map."""
+
+    def __init__(
+        self,
+        source_images: Sequence[Dataset],
+        pixel_array: np.ndarray,
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        contains_recognizable_visual_features: bool,
+        real_world_value_mappings: Sequence[RealWorldValueMapping],
+        window_center: Union[int, float],
+        window_width: Union[int, float],
+        transfer_syntax_uid: Union[str, UID] = ImplicitVRLittleEndian,
+        content_description: Optional[str] = None,
+        content_creator_name: Optional[str] = None,
+        pixel_measures: Optional[PixelMeasuresSequence] = None,
+        plane_orientation: Optional[PlaneOrientationSequence] = None,
+        plane_positions: Optional[Sequence[PlanePositionSequence]] = None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        source_images: Sequence[pydicom.Dataset]
+            Sequence of one or more `pydicom.Dataset`s that this parametric map
+            is derived from
+        pixel_array: numpy.ndarray
+            Array of parametric map pixel data of unsigned integer or
+            floating-point data type representing one or more frames of the
+            parametric map pixel data. The values are supposed to represent a
+            single "feature", i.e., be the result of one set of image
+            transformations such that the same `real_world_value_mappings`
+            apply.
```

Comment:
I think more explanation is probably needed here about the shape/dimensions, what order they come in and how they need to relate the the source images (or plane positions)